File: src/main/java/me/bristermitten/plumber/scheduling/ScheduledTask.java
package me.bristermitten.plumber.scheduling;

import me.bristermitten.plumber.aspect.AspectAnnotation;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@AspectAnnotation(target = SchedulerAspect.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface ScheduledTask {
}

File: src/main/kotlin/me/bristermitten/plumber/scheduling/ScheduledTask.kt
package me.bristermitten.plumber.scheduling

import me.bristermitten.plumber.aspect.AspectAnnotation

/**
 * Aspect annotation linking to [SchedulerAspect]
 * Currently, does nothing as the scheduler aspect is always loaded
 */
@AspectAnnotation(target = SchedulerAspect::class)
@Retention(AnnotationRetention.RUNTIME)
annotation class ScheduledTask
